scalameta/contrib/shared/src/main/scala/scala/meta/contrib/implicits/ReplaceExtensions.scala

```diff
@@ -0,0 +1,12 @@
+package scala.meta.contrib.implicits
+
+import scala.collection.immutable.Seq
+import scala.meta.contrib._
+
+trait ReplaceExtensions {
+  implicit class ExtensionReplacers[A](a: A) {
+    def replace[B](bs: Seq[B])(implicit ev: Replace[A, B]): A = ev.replace(a, bs)
```

Comment:
@ShaneDelmore @olafurpg I mostly agree with you both.

Heres my take. (Probably because I write 90% Java code at $JOB)

x.setY = mutate the Y property of X
x.withY = return a new X with the property Y changed
x.replace = replace without caring about current value (as @ShaneDelmore said)
x.map = apply a function A => B. (For each element A, give me a B)
x.transform = map for tree's (Possibly????)
x.update = same as map.

In this PR, I will add `withStats` and `withMods` as @olafurpg recommended. If you disagree with the term `with`, I'm open for discussion.